### PR TITLE
LibGUI: Fix TextEditor crash when double clicking an empty line.

### LINKED ISSUE #5995: Text Editor: Crash on double clicking on an empty line
Crash only happens in `.js`, `.cpp`, `.h` (these are the one I tested) files. Other files like `.frm`, `Makefile` and `.md` are fine.
Doesn't crash in HackStudio.
Stack trace:
```
CrashDaemon(18): New coredump file: /tmp/coredump/TextEditor_54_1616921537
CrashDaemon(18): --- Backtrace for thread #0 (TID 54) ---
CrashDaemon(18): 0x5acab1a8: [libsystem.so]  +0x8 (syscall.cpp:35)
CrashDaemon(18): 0xb1f6dcc6: [libc.so] __assertion_failed +0x106 (syscall.h:44)
CrashDaemon(18): 0x624f1b7f: [libgui.so] GUI::TextDocument::text_in_range(GUI::TextRange const&) const +0x2bf (Utf32View.h:91)
CrashDaemon(18): 0x624f73fe: [libgui.so] GUI::TextEditor::selected_text() const +0x7e (TextEditor.cpp:1235)
CrashDaemon(18): 0x812072ac: [/bin/TextEditor] TextEditorWidget::update_statusbar() +0x1cc (Atomic.h:266)
CrashDaemon(18): 0x8120768d: [/bin/TextEditor] AK::Function<void ()>::CallableWrapper<TextEditorWidget::TextEditorWidget()::{lambda()#17}>::call() const +0x1d (Function.h:111)
CrashDaemon(18): 0x624fb4b3: [libgui.so] GUI::TextEditor::set_cursor(GUI::TextPosition const&) +0xe3 (Function.h:57)
CrashDaemon(18): 0x624fbeed: [libgui.so] GUI::TextEditor::doubleclick_event(GUI::MouseEvent&) +0x1bd (TextEditor.cpp:256)
CrashDaemon(18): 0x6251ae9a: [libgui.so] .L2872 +0xa (Widget.cpp:271)
CrashDaemon(18): 0x78277c84: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(18): 0x62528368: [libgui.so] GUI::Window::handle_mouse_event(GUI::MouseEvent&) +0x5b8 (Window.cpp:375)
CrashDaemon(18): 0x6252aa19: [libgui.so] GUI::Window::event(Core::Event&) +0x39 (Window.cpp:545)
CrashDaemon(18): 0x78277c84: [libcore.so] Core::Object::dispatch_event(Core::Event&, Core::Object*) +0x64 (Object.h:104)
CrashDaemon(18): 0x78260ff4: [libcore.so] Core::EventLoop::pump(Core::EventLoop::WaitMode) +0x3e4 (Atomic.h:266)
CrashDaemon(18): 0x782615bb: [libcore.so] Core::EventLoop::exec() +0x4b (EventLoop.cpp:390)
CrashDaemon(18): 0x62438d65: [libgui.so] GUI::Application::exec() +0x25 (Application.cpp:121)
CrashDaemon(18): 0x81201696: [/bin/TextEditor] main +0x496 (main.cpp:106)
CrashDaemon(18): 0x81201a05: [/bin/TextEditor] _start +0x45 (crt0_shared.cpp:60)
```

### PR DESCRIPTION
**Pull request by Kubiisek**
I also reduced the number of calls to `TextEditor::did_update_selection` - not sure if I should make a second PR. Fixes #5995

### CODE REVIEW

**Comment by linusg** [comment]
@Kubiisek you have a typo in the description, #5997 is this PR itself and should be #5995.

**Comment by awesomekling** [comment]
Please wrap your commit messages at 72 characters.

**Comment by Kubiisek** [comment]
My bad, it looked like the IDE did it for me. Anyway it's fixed.

**Comment by Kubiisek** [comment]
Any other feedback?

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 1]
```diff
@@ -713,6 +713,7 @@ void TextEditor::paint_event(PaintEvent& event)
 
 void TextEditor::select_all()
 {
+    dbgln("select_all");
```
Comment (awesomekling) [inline Userland/Libraries/LibGUI/TextEditor.cpp:716]: This seems accidental :)

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
```
Comment (awesomekling) [inline Userland/Libraries/LibGUI/TextEditor.cpp:2]: Please don't modify other people's copyright lines. Feel free to add your own instead :)

File: Userland/Libraries/LibGUI/TextDocument.cpp [hunk 3]
```diff
@@ -352,6 +352,8 @@ String TextDocument::text_in_range(const TextRange& a_range) const
         auto& line = this->line(i);
         size_t selection_start_column_on_line = range.start().line() == i ? range.start().column() : 0;
         size_t selection_end_column_on_line = range.end().line() == i ? range.end().column() : line.length();
+        if (selection_start_column_on_line == selection_end_column_on_line)
+            return String("");
```
Comment (awesomekling) [inline Userland/Libraries/LibGUI/TextDocument.cpp:356]: The idiomatic way to get an empty `String` is usually `String::empty()` :)

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Comment by Kubiisek** [comment]
Thanks, I fixed all of your issues :+1:

File: Userland/Libraries/LibGUI/TextEditor.cpp [hunk 4]
```diff
@@ -1,5 +1,6 @@
 /*
- * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
+ * 2021, the SerenityOS developers
```
Comment (awesomekling) [inline Userland/Libraries/LibGUI/TextEditor.cpp]: Please use the same format as other files in the project:

```
 * Copyright (c) 2018-2020, Andreas Kling <kling@serenityos.org>
 * Copyright (c) 2021, the SerenityOS developers
```

**Review by awesomekling — CHANGES_REQUESTED** [review]
Please observe the [contribution guidelines](https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md), specifically
>Squash your commits when making revisions after a patch review.

**Comment by Kubiisek** [comment]
I'm really sorry for the force pushes. Thank you for your patience with me.

**Comment by linusg** [comment]
Already fixed by #6537.

**Pull request closed without merging**